lib/cadet_web/views/grading_view.ex

```diff
@@ -35,17 +35,26 @@ defmodule CadetWeb.GradingView do
   def render("grading_info.json", %{answer: answer}) do
     transform_map_for_view(answer, %{
       student: &transform_map_for_view(&1.submission.student, [:name, :id]),
-      question:
-        &Map.put(
-          CadetWeb.AssessmentsView.build_question(%{question: &1.question}),
-          :answer,
-          &1.answer["code"] || &1.answer["choice_id"]
-        ),
+      question: &build_grading_question/1,
       solution: &(&1.question.question["solution"] || ""),
       grade: &build_grade/1
     })
   end
 
+  defp build_grading_question(answer) do
+    results =
+      case answer.autograding_results do
```

Comment:
i think using pattern matching here would be more expressive!! something like

```elixir
defp build_autograding_results(nil), do: nill
defp build_autograding_results(results) do
    Enum.map(results, &CadetWeb.AssessmentsView.build_result/1)
end
```

then just call the function below or put it in a variable first also can!!

Reply:
Refactored as requested.